src/report_reader.cpp

```diff
@@ -278,79 +298,112 @@ std::string ReportReader<T>::Population::getDataUnits() const {
 
 template <typename T>
 bool ReportReader<T>::Population::getSorted() const {
-    return nodes_ids_sorted_;
+    return is_node_ids_sorted_;
 }
 
 template <typename T>
 std::vector<NodeID> ReportReader<T>::Population::getNodeIds() const {
-    return nodes_ids_;
+    return node_ids_;
 }
 
 template <typename T>
 typename ReportReader<T>::Population::NodeIdElementLayout
 ReportReader<T>::Population::getNodeIdElementLayout(
-    const nonstd::optional<Selection>& node_ids) const {
-    std::vector<NodeID> concrete_node_ids;
+    const nonstd::optional<Selection>& node_ids,
+    const nonstd::optional<size_t>& _block_gap_limit) const {
     NodeIdElementLayout result;
-    result.min_max_range = {std::numeric_limits<uint64_t>::max(),
-                            std::numeric_limits<uint64_t>::min()};
+    std::vector<NodeID> concrete_node_ids;
     size_t element_ids_count = 0;
 
-    // Helper function to update layout, alongside the min / max values and count
-    const auto update_layout = [&](const NodeID& node_id, const Selection::Range& range) {
-        concrete_node_ids.emplace_back(node_id);
-        result.node_ranges.emplace_back(range);
+    // Set the gap between IO blocks while fetching data (Default: 64MB / 4 x GPFS blocks)
+    const size_t block_gap_limit = _block_gap_limit.value_or(16777216);
 
-        result.min_max_range.first = std::min(result.min_max_range.first, range.first);
-        result.min_max_range.second = std::max(result.min_max_range.second, range.second);
-        element_ids_count += (range.second - range.first);
-    };
+    if (block_gap_limit < 4194304) {
```

Comment:
No, not exactly. Each GPFS block is 16MB, so having a limit matching the block-size is reasonable, because anyway if the gap is smaller than that, you are most probably going to fetch the whole block anyway.

If we leave this a bit open, for instance by allowing users to set `0` or something small, then we would be in the same situation like we were when we started doing this whole `{min,max}` optimization (i.e., `libsonata` triggering millions of IO transactions). Therefore, I'd prefer if we could set a magic number limit, like 1 GPFS block, to at least limit how low you can set this parameter.

Reply:
I'm really hoping no-one ever tweaks the parameter in the first place, but that's an aside.

I'm just wondering if we're making it too GPFS centric; but I guess the same value makes sense on an NVME node?

Reply:
I think it makes sense for other systems, still. Even on NVMEs, we might want to have larger reads for lower latency…

Reply:
As @matz-e says, it shouldn't matter because here we would just force to use the `{min,max}` optimization as much as possible. For the NVMe drives, the higher the requests also the better to reduce the latency.